### DISCUSSION TITLE
Type error: Cannot find module '@/app/ui/fonts' or its corresponding type declarations.

### DISCUSSION THREAD

**Discussion by ozee1z** [vercel/next.js#60173] [Help] [answered]
### Summary

I tried to host NEXT JS project on vercel with github repo. I followed the tutorials from NEXT JS official sites but seeing this error message. ( Type error: Cannot find module '@/app/ui/fonts' or its corresponding type declarations.

  

### Additional information

```js
1 | import { GlobeAltIcon } from '@heroicons/react/24/outline';
> 2 | import { lusitana } from '@/app/ui/fonts';
    |                          ^
  3 |
  4 | export default function AcmeLogo() {
  5 |   return (
Error: Command "npm run build" exited with 1 )
```

### Example

https://vercel.com/schlinks-projects/nextjs-dashboard/A9ooANdwGFCA1nLFQ8ZHZC2tvZMe

**Comment by ananuness** [comment]
Hi! Could you show your `app/ui/fonts` folder and your `tsconfig.json` please?

**Reply by ozee1z** [reply]
![app-ui-fonts](https://github.com/vercel/next.js/assets/48673395/0d04c8fb-cd2d-43f4-96b8-590e0bcea850)
![tsconfig](https://github.com/vercel/next.js/assets/48673395/daeea6f6-fa85-47c0-8138-59bb35cffc0e)

**Reply by ananuness** [reply]
Oh, how are you using aliases without defining them in the `tsconfig`? 😳 I really thought we needed them to configure it. Oh, and I meant your `ui/fonts` folder structure ☺️

**Reply by ananuness** [reply]
But, if you want to map your path aliases, you can define a `baseUrl`, generally is the root folder and then you can set your path aliases like that:

![Captura de tela de 2024-01-04 08-01-32](https://github.com/vercel/next.js/assets/51466624/f22a78dd-99ff-4e5f-9416-ec1badae6944)

As you can see in `@/clientComponents` I just set the path without `*` because it's a index file, but if not, you can generalize to call any file in your folders!

**Reply by ozee1z** [reply]
![image](https://github.com/vercel/next.js/assets/48673395/717f2075-9d13-4b5b-b0fa-c3c19b2c83ee)

How do I define the tsconfig? I followed everything the tutorial said.

**Reply by ananuness** [reply, marked as answer]
Okay! So let's start defining the `baseUrl` and the `paths` in the **tsconfig** as you can see in my previous answer! You can define:

```
"baseUrl": ".", 
paths: {
  "@/app/*": ["app/*"]
}
```

And see if it works for you already!

**Reply by ozee1z** [reply]
It worked this was the only code I added to the tsconfig: 
"baseUrl": "*", 
paths: {
  "@/app/*": ["app/*"]
}

Thanks Ana.

**Reply by ananuness** [reply]
You're welcome!! And as you can see in the image in the previous comment, you can add as many path aliases as you wish! Have a great day!

**Reply by Lutir** [reply]
I got the same error! Thanks Ana and Schlink! The fix worked for me

**Reply by ananuness** [reply]
I'm glad it helped you!!

**Reply by jhorton84** [reply]
@ananuness I'm getting this same error, but when I try the above fix I get the error for all imports using '@/app/ui/fonts';

**Reply by ananuness** [reply]
@jhorton84 could you upload the pic again? It didn't load completely!

**Reply by jhorton84** [reply]
<img width="1728" alt="Screenshot 2024-01-06 at 3 43 58 PM" src="https://github.com/vercel/next.js/assets/42826280/65ce5b5f-e884-4c01-bc46-3e18c6168367">

**Reply by jhorton84** [reply]
<img width="292" alt="Screenshot 2024-01-06 at 7 50 27 PM" src="https://github.com/vercel/next.js/assets/42826280/8eff8845-7ffc-4313-b799-8f4cb5f76775">

**Reply by jhorton84** [reply]
I don't know if these screenshots provide enough information, but I'm thoroughly stumped on how to resolve this issue. Let me know if I can provide any other information such as versions of React, NPM, Node, etc that could be affecting this. Thanks!

**Reply by Lutir** [reply]
Try out this in the `tsconfig.json`

```
...
    ],
    "baseUrl": "",
    "paths": {
      "@/app/*": ["app/*"]
    }
...
```

No other change needed

**Reply by OstrowskiDev** [reply]
Had the same problem, thx for providing solution. This worked for me:

```
...
    ],
    "baseUrl": "",
    "paths": {
      "@/app/*": ["app/*"]
    }
...
```

**Reply by ananuness** [reply]
@jhorton84 it's because of `@/*`, as you can see, when you define the `baseUrl` you already provided the start point, so you don't need to set `./*` instead you can try `/*`.

But the advantage of using path aliases is to reduce the path, make it more clean!! So it would be nice if you define a path alias to your main folders in `/app`. Just remove this alias (`"@/*": ["./*"]`) and see if it works!

**Reply by jhorton84** [reply]
I'm going to delete the repo and go back through the course again from the beginning making the change to the tsconfig immediately as shown in the example and see if it works at the build step in chapter 6.
Thanks for the responses!

**Reply by letsclode** [reply]
"baseUrl": "*",
    "paths": {
        "@/app/*": ["app/*"]
      }
      
      what i did replace this in tsconfig.json and then re import every fail imports

**Reply by ananuness** [reply]
@Revitalismo  I'm sorry if the solution didn't worked for you as the problem of the discussion was solved by just adding import aliases. Also, I corrected a typo of mine! I had typed an asterisk as `baseUrl`, but it was supposed to be a **dot** to indicate that the base is the root of the project or it could be the path of your choice.

**Reply by kristinakisterherrick** [reply]
@jhorton84, make sure you add the baseUrl and paths that @ananuness showed, below the exclude item, not inside the compilerOptions item. I'll attach my entire tsconfig.json file for reference.
[tsconfig.json](https://github.com/vercel/next.js/files/14350883/tsconfig.json)

**Reply by chabandou** [reply]
I have the same problem, tried all the provided solutions none worked, just wanted to deploy the code from the official course,
here's the repo for the app:
https://github.com/chabandou/nextjs-dashboard

**Reply by chabandou** [reply]
I found a fix after searching online and checking [TS Docs](https://www.typescriptlang.org/tsconfig#baseUrl),
The baseUrl should be "./" to include all the folders/files in the same root as tsconfig.json
which makes the file look like this:
```
{
  "compilerOptions": {
    ... //,
    "baseUrl": "./",
    "paths": {
      "@/*": ["./*"]
    }
  },
 ... //
}
```

**Reply by eatthemenu** [reply]
> Okay! So let's start defining the `baseUrl` and the `paths` in the **tsconfig** as you can see in my previous answer! You can define:
> 
> ```
> "baseUrl": ".", 
> paths: {
>   "@/app/*": ["app/*"]
> }
> ```
> 
> And see if it works for you already!

Hello @ananuness , I have questions regarding your solution.

1. Are the baseUrl and paths fields the children of "compilerOptions"?  Or are they supposed to be siblings to "compilerOptions"?
2. On Vercel.com, when I "deploy" I get the error "Module not found: Can't resolve '@/app/ui/fonts'" just like others.  What I don't understand as a newbie is why this error does NOT occur in my local environment but does on Vercel.com.
3. Also, some people say that the "baseUrl" value should be "" and not ".".  Some other say it should be "./".  Just double checking, the correct value is "."?
4. The label paths should be surround by double quotes, correct?  As in "paths": {"@/app/*":["app/*"]}.

**Reply by ananuness** [reply]
Hello @eatthemenu! sorry the delay, let's go to some answers:
1. Yes, both of them are children of `compilerOptions`
2. I'm not 100% sure of it, but I think it can occur because its build! Depending on how you're setting the baseUrl, for example, it can change where it's pointing: local -> it's a defined folder, build -> can be another one, like the `dist`, also, maybe your changes didn't reflect on vercel, it's nice to take a look!
3. The `baseUrl` defines which is your start point! For example, you have a project inside the folder `react` , if you set the base to `.` , then the path `react/src/...` is the same as `src/...` in your paths property (that's why you don't need to repeat the root folder name here). When it's `./` it's the same as `.` as it points to the same content of the root folder, but here you may have a problem if you need to import a root file, because in paths it will be like `.file_name` (wrong way to import a file, right?), so if you define the path `"@/*": ["./*"]` it will need the base as `./` so you can import anything and when it's an empty string, you just need to define the full path when setting the `paths` property, so to resume: **baseUrl is the property you define the start point to your paths!**
 4. Yes, and don't forget to add the `*` after the `app/` , because it's the way of saying that it includes all paths inside your `app` folder, so it would be right this way: "@/app/*": ["app/*"] now it's possible to have a path like @/app/components/... and so on!
 
 I hope it helps!! 🩵

**Reply by Jarabi** [reply]
Thanks ananuness

**Comment by seanodotcom** [comment]
This fix worked for me. It should be mentioned in the official Vercel NextJS Tutorial | Chapter 6.

> I found a fix after trying a few times and checking [TS Docs](https://www.typescriptlang.org/tsconfig#baseUrl), The baseUrl should be "./" to include all the folders/files in the same root as tsconfig.json which makes the file look like this:
> 
> ```
> {
>   "compilerOptions": {
>     ... //,
>     "baseUrl": "./",
>     "paths": {
>       "@/*": ["./*"]
>     }
>   },
>  ... //
> }
> ```

**Reply by milkion** [reply]
Thank you so much, it works!

**Comment by Oms-343** [comment]
If nothing works , do check the file name  for the fonts.tsx  you create . I missed  's' and everywhere i was importing  the wrong filename.

**Reply by reflextoogood** [reply]
Dude I legit made the same mistake

**Reply by guillim** [reply]
can't believe i did the same shit 😆 . Thanks for pointing out

**Comment by ibrahim-sowunmi** [comment]
I ran into an error here because I skipped Chapter 3 folks!

**Reply by whenmoon** [reply]
Hahaha same but then did we do the Next JS tutorial to learn how to use some styles package? Nope

**Comment by Joachim92** [comment]
In my case I was getting this error because I decided to build the project before completing the first part of the tutorial. This file [dashboard/final-example/app/ui/fonts.ts](https://github.com/vercel/next-learn/blob/main/dashboard/final-example/app/ui/fonts.ts) was missing and I now realize that it was intended to be created in a later stage in the tutorial.

**Comment by seyedahmaddev** [comment]
i have problem this problem so:
<html>
<body>

Failed to compile.
--
16:57:29.215 |  
16:57:29.215 | ./app/ui/acme-logo.tsx:2:26
16:57:29.215 | Type error: Cannot find module '@/app/ui/fonts' or its corresponding type declarations.
16:57:29.216 |  
16:57:29.216 | 1 \| import { GlobeAltIcon } from '@heroicons/react/24/outline';
16:57:29.216 | > 2 \| import { lusitana } from '@/app/ui/fonts';
16:57:29.216 | \|                          ^
16:57:29.216 | 3 \|
16:57:29.216 | 4 \| export default function AcmeLogo() {
16:57:29.218 | 5 \|   return (
16:57:29.277 | Error: Command "npm run build" exited with 1


</body>
</html>
-------------------
But I noticed that the changes that I make in my code in Visual Studio or GitHub directly, are not given in the Versel code. delete yout project in vercel and then reinstall new project from corrected github files.

**Reply by romu70** [reply]
Many many thanks !!
I was going totally mad at looking at this issue. And as a newbie, my mind was upside down at knowing what to do.

So yes:
1. Fix the path
2. Delete the Vercel project
3. Import it again

**Comment by nidhinvs78** [comment]
You may need to install the package google fonts.
try this, and push the changes to github.
`npm i next-google-fonts`

**Comment by mongolsteppe** [comment]
Created a pull request to avoid future users running into this issue https://github.com/vercel/next-learn/pull/674

**Reply by beckrecca** [reply]
It'd be great if someone approved this PR / edited Chapter 6 accordingly because when you install the linter in Chapter 14 (https://nextjs.org/learn/dashboard-app/improving-accessibility), it overwrites the tsconfig file and breaks routing completely until you re-apply this fix to that file.

**Comment by realtebo** [comment]
please fix example app project or, better, add a few lines to the documentations of learning project to help newbies to find and fix this error.

**Comment by wcilzo** [comment]
In next 14, you can see https://github.com/vercel/next-learn/issues/580#issuecomment-2065616306, I have solved the problem.😊

**Comment by HeadStone7** [comment]
Just create a file in  '@/app/ui' name "fonts.ts" and then add: 

``
import { Inter, Lusitana } from 'next/font/google';
export const inter = Inter({ subsets: ['latin'] });
export const lusitana = Lusitana({
                   weight: ['400', '700'],
                   subsets: ['latin'],
                   }); 
``
Have a great day !

**Comment by YuraniDanilo** [comment]
Hello guys, I found a solution to that problem: 

If you're using windows, your system is identifying FONTS.TS as a video MPEG-2 not as a Typescript file. You must change it. If you look inside your Github project, you won't find FONTS.TS there. Because it's not importing because of this problem. 

Open the properties of fonts.ts and put webstorm or vs code as default and then import it inside the Github repository. It will work.

**Comment by IsacFSC** [comment]
Ao tentar fazer deploy no vercel do meu projeto com nextjs, prisma e neon está dando erro mas não encontrei no meu código. 

**ERRO NO VERCEL ABAIXO:**

> 1. ./src/app/admin/dashboard/page.tsx
> 2. 10:38  Error: React Hook "useAuth" is called in function "adminDashboardPage" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use".  react-hooks/rules-of-hooks
